docs/tasks-overview.md

```diff
@@ -1,45 +1,62 @@
-| **{configName}Benchmark**                          | Executes all benchmarks for the specified Wasm target.                                                                                         |
-| **{configName}BenchmarkGenerate**                  | Generates Wasm source files for the specified Wasm target. These source files will be used in the benchmarking process.                        |
-
-## Kotlin/Native Specific Tasks
-
-| Task                                                       | Description                                                                                                                                                                           |
-| ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **link{configName}BenchmarkReleaseExecutable{configName}** | Compiles the generated files and creates an executable. The entry point for the executable is the generated main function.                                                            |
-| **{configName}Benchmark**                                  | Executes the benchmarks for each benchmark configuration defined in the plugin extension corresponding to the specific config. For the "main" configuration, `configName` is dropped. |
-| **{configName}BenchmarkGenerate**                          | Takes compiled user code, retrieves metadata and generates the code needed for measurement. This is a native-target-specific task (e.g., for `macosX64()`, `native` -> `macosX64`).   |
+## Overview of Tasks Provided by kotlinx-benchmark Gradle Plugin
+
+The kotlinx-benchmark plugin creates different Gradle tasks depending on how it is configured.
+For each pair of configuration profile and registered target a task is created to execute that profile on the respective platform.
+To learn more about configuration profiles, refer to [configuration-options.md](configuration-options.md).
+
+### Example Configuration
+
+To illustrate, consider the following `kotlinx-benchmark` configuration:
+
+```kotlin
+// build.gradle.kts
+benchmark {
+    configurations {
+        named("main") {
+            iterations = 20
+            warmups = 20
+            iterationTime = 1
+            iterationTimeUnit = "s"
+        }
+        register("smoke") {
+            include("Essential")
+            iterations = 10
+            warmups = 10
+            iterationTime = 200
+            iterationTimeUnit = "ms"
+        }
+    }
+
+    targets {
+        register("jvm")
+        register("js")
+    }
+}
+```
+
+## Tasks for the "main" Configuration Profile
+
+- **`benchmark`**:
+    - Runs benchmarks within the "main" profile for all registered targets.
```

Comment:
Could you please also add that "In our example, `benchmark` runs benchmarks within the "main" profile in both `jvm` and `js` targets."

Reply:
Added